### ISSUE TITLE
Feature: kill all chrome zombie process upon browser close with treekill

### ISSUE THREAD

**Issue by sandro-salles** [open-wa/wa-automate-nodejs#563]
I've noticed a degradation in system resources usage when connecting/disconnecting multiple sessions over time, with some ghost/zombie chrome child processes kept running even after calling client.kill (or when a auth/qr timeout has happend).

While searching for a way to help deal with those processes i [found this very interesting article](https://docs.browserless.io/blog/2019/03/13/more-observations.html#tip-2-when-youre-done-kill-it-with-fire), and implemented on my code and it really helps a lot.

Basically, after a successful connection i'm storing the browser pid with **(await client.browser()).process().pid** and after **await client.kill()** i call  **treekill(browser_pid, 'SIGKILL')** and it finishes all zombies...

Maybe adding this every time the code is closing the browser can help reduce unnecessary resources usage..

Regards

**Comment by smashah** [comment]
Wow that's actually very useful thanks for the find @sandro-salles . Will implement asap

**Comment by sandro-salles** [comment]
One more interesting finding, in order to reduce the amount of chrome helper processes running (sometimes it spawns 6 processes for each browser instance opened), i'm setting the following chromium flags:

```
[
  '--disable-2d-canvas-clip-aa',
  '--disable-2d-canvas-image-chromium',
  '--disable-3d-apis',
  '--disable-accelerated-2d-canvas',
  '--disable-accelerated-jpeg-decoding',
  '--disable-accelerated-mjpeg-decode',
  '--disable-accelerated-video-decode',
  '--disable-app-list-dismiss-on-blur',
  '--disable-audio-output',
  '--disable-background-timer-throttling',
  '--disable-backgrounding-occluded-windows',
  '--disable-breakpad',
  '--disable-canvas-aa',
  '--disable-client-side-phishing-detection',
  '--disable-component-extensions-with-background-pages',
  '--disable-composited-antialiasing',
  '--disable-default-apps',
  '--disable-dev-shm-usage',
  '--disable-extensions',
  '--disable-features=TranslateUI,BlinkGenPropertyTrees',
  '--disable-field-trial-config',
  '--disable-fine-grained-time-zone-detection',
  '--disable-geolocation',
  '--disable-gl-extensions',
  '--disable-gpu',
  '--disable-gpu-early-init',
  '--disable-gpu-sandbox',
  '--disable-gpu-watchdog',
  '--disable-histogram-customizer',
  '--disable-in-process-stack-traces',
  '--disable-infobars',
  '--disable-ipc-flooding-protection',
  '--disable-notifications',
  '--disable-renderer-backgrounding',
  '--disable-session-crashed-bubble',
  '--disable-setuid-sandbox',
  '--disable-site-isolation-trials',
  '--disable-software-rasterizer',
  '--disable-sync',
  '--disable-threaded-animation',
  '--disable-threaded-scrolling',
  '--disable-translate',
  '--disable-webgl',
  '--disable-webgl2',
  '--enable-features=NetworkService',
  '--force-color-profile=srgb',
  '--headless',
  '--hide-scrollbars',
  '--ignore-certifcate-errors',
  '--ignore-certifcate-errors-spki-list',
  '--ignore-certificate-errors',
  '--ignore-certificate-errors-spki-list',
  '--ignore-gpu-blacklist',
  '--ignore-ssl-errors',
  '--log-level=3',
  '--metrics-recording-only',
  '--mute-audio',
  '--no-crash-upload',
  '--no-default-browser-check',
  '--no-experiments',
  '--no-first-run',
  '--no-sandbox',
  '--no-zygote',
  '--renderer-process-limit=1',
  '--safebrowsing-disable-auto-update',
  '--silent-debugger-extension-api',
  '--single-process',
  '--unhandled-rejections=strict',
  '--window-position=0,0'
]
```

Some of them are present on **puppeteerConfig. chromiumArgs** , but i think that specially the following ones really help on that:

```
  '--single-process',
  '--no-zygote',
  '--renderer-process-limit=1',
  '--no-first-run',
```

I hope it can help someone else!

Regards!

**Comment by smashah** [comment]
c710c1638743a7d6d79a4145b592f0842c3e23d6

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i`);
exec(`npm run release-ci`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [1.9.80](https://github.com/open-wa/wa-automate-nodejs/compare/1.9.79...1.9.80) (2020-06-24)

- Release 1.9.80 [`1c3018c`](https://github.com/open-wa/wa-automate-nodejs/commit/1c3018c254677ca3eb73a8131f0082eb5d9bca2d)
- Feature: Perf improvements and zombie process killer [`c710c16`](https://github.com/open-wa/wa-automate-nodejs/commit/c710c1638743a7d6d79a4145b592f0842c3e23d6)

**Comment by smashah** [comment]
Please 1.9.81 for the changes. Thanks @sandro-salles

**Comment by sandro-salles** [comment]
@smashah maybe the kill on initializer should have this same logic?

https://github.com/open-wa/wa-automate-nodejs/blob/e428aa8e2f8de559bcaa2672a87e16d778bda802/src/controllers/initializer.ts#L208
